test/test_saveat_solution.py

```diff
@@ -147,6 +147,42 @@ def test_saveat_solution():
     assert sol.result == diffrax.RESULTS.successful
 
 
+@pytest.mark.parametrize("subs", [True, False])
+def test_t0_eq_t1(subs):
```

Comment:
This test LGTM! Can you also add a variant in which the computation is `vmap`d? (So that e.g. one batch element has `t0 == t1` and another one doesn't, or multiple batch elements have `t0 == t1` but with different such values.)

This test will be sure that the `eqxi.unvmap_any` trick I mention above is handled correctly.